machine_learning/loss_functions/hinge_loss.py

```diff
@@ -0,0 +1,55 @@
+"""
+Hinge Loss
+
+Description:
+Compute the Hinge loss used for training SVM (Support Vector Machine).
+
+Formula:
+loss = max(0, 1 - true * pred)
+
+Reference: https://en.wikipedia.org/wiki/Hinge_loss
+
+Author: Poojan Smart
+Email: smrtpoojan@gmail.com
+"""
+
+import numpy as np
+
+
+def hinge_loss(y_true: np.ndarray, pred: np.ndarray) -> float:
+    """
+    Calculate the hinge loss for y_true and pred for binary classification.
```

Comment:
```suggestion
    Calculate the mean hinge loss for y_true and pred for binary classification.
```
The documentation should reflect the fact that you're averaging the hinge losses over the arrays